shared/XrSceneLib/BgfxRenderer.cpp

```diff
@@ -0,0 +1,175 @@
+//
+//*********************************************************
+#pragma once
+
+#include "pch.h"
+#include "BgfxRenderer.h"
+
+using namespace engine;
+
+namespace sample::bg {
+
+    struct CachedFrameBuffer {
+        std::vector<unique_bgfx_handle<bgfx::FrameBufferHandle>> FrameBuffers;
+        winrt::com_ptr<ID3D12CommandAllocator> CommandAllocator;
+    };
+
+    std::map<std::tuple<void*, void*>, CachedFrameBuffer> s_cachedFrameBuffers;
+
+    void RenderView(const XrRect2Di& imageRect,
+                    const float renderTargetClearColor[4],
+                    const std::vector<xr::math::ViewProjection>& viewProjections,
+                    DXGI_FORMAT colorSwapchainFormat,
+                    void* colorTexture,
+                    DXGI_FORMAT depthSwapchainFormat,
+                    void* depthTexture
+                    ,const std::vector<std::unique_ptr<Scene>>& activeScenes,
+                    const FrameTime& frameTime, 
+                    bool& submitProjectionLayer
+    ) {
+        
+        //static int frameCount = 0;
+        //wchar_t text_buffer[20] = {0};                                               // temporary buffer
+        //swprintf(text_buffer, _countof(text_buffer), L"frame: %d \n", ++frameCount); // convert
+        //OutputDebugString(text_buffer);                                              // print
+
+        const uint32_t viewInstanceCount = (uint32_t)viewProjections.size();
+        /* CHECK_MSG(viewInstanceCount <= CubeShader::MaxViewInstance,
+                   "Sample shader supports 2 or fewer view instances. Adjust shader to accommodate more.");*/
+
+        const bool reversedZ = viewProjections[0].NearFar.Near > viewProjections[0].NearFar.Far;
+        const float depthClearValue = reversedZ ? 0.f : 1.f;
+        const uint64_t state =
+            BGFX_STATE_WRITE_MASK | (reversedZ ? BGFX_STATE_DEPTH_TEST_GREATER : BGFX_STATE_DEPTH_TEST_LESS) | BGFX_STATE_CULL_CCW;
+
+        auto iter = s_cachedFrameBuffers.find({colorTexture, depthTexture});
+        if (iter == s_cachedFrameBuffers.end()) {
+            unique_bgfx_handle<bgfx::TextureHandle> colorTex(
+                bgfx::createTexture2D(1,
+                                      1,
+                                      false,
+                                      (uint16_t)viewInstanceCount,
+                                      DxgiFormatToBgfxFormat(colorSwapchainFormat),
+                                      (IsSRGBFormat(colorSwapchainFormat) ? BGFX_TEXTURE_SRGB : 0) | BGFX_TEXTURE_RT));
+            unique_bgfx_handle<bgfx::TextureHandle> depthTex(bgfx::createTexture2D(
+                1, 1, false, (uint16_t)viewInstanceCount, DxgiFormatToBgfxFormat(depthSwapchainFormat), BGFX_TEXTURE_RT_WRITE_ONLY));
+
+            // Force BGFX to create the texture now, which is necessary in order to use overrideInternal.
+            bgfx::frame();
+            auto x = colorTex.get();
```

Comment:
Are there better variable names possible for x,y? Or is this the standard naming for these type of vars